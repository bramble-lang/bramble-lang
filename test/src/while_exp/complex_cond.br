fn my_main() -> i64 {
    let mut x: i64 := 0;
    let mut y: i64 := 15;
    while (x <= 5 && y >= 11) {
        project::std::io::writei64ln(x);
        project::std::io::writei64ln(y);
        mut x := x + 1;
        mut y := y - 1;
    };
    return 0;
}