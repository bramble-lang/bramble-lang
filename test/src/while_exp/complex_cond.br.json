[{"id": 1, "stage": "lexer", "source": [0, 2], "ok": "Keyword"},
{"id": 2, "stage": "lexer", "source": [3, 10], "ok": "Identifier"},
{"id": 3, "stage": "lexer", "source": [10, 11], "ok": "Operator"},
{"id": 4, "stage": "lexer", "source": [11, 12], "ok": "Operator"},
{"id": 5, "stage": "lexer", "source": [13, 15], "ok": "Operator"},
{"id": 6, "stage": "lexer", "source": [16, 19], "ok": "Primitive"},
{"id": 7, "stage": "lexer", "source": [20, 21], "ok": "Operator"},
{"id": 8, "stage": "lexer", "source": [26, 29], "ok": "Keyword"},
{"id": 9, "stage": "lexer", "source": [30, 33], "ok": "Keyword"},
{"id": 10, "stage": "lexer", "source": [34, 35], "ok": "Identifier"},
{"id": 11, "stage": "lexer", "source": [35, 36], "ok": "Operator"},
{"id": 12, "stage": "lexer", "source": [37, 40], "ok": "Primitive"},
{"id": 13, "stage": "lexer", "source": [41, 43], "ok": "Operator"},
{"id": 14, "stage": "lexer", "source": [44, 45], "ok": "Integer"},
{"id": 15, "stage": "lexer", "source": [45, 46], "ok": "Operator"},
{"id": 16, "stage": "lexer", "source": [51, 54], "ok": "Keyword"},
{"id": 17, "stage": "lexer", "source": [55, 58], "ok": "Keyword"},
{"id": 18, "stage": "lexer", "source": [59, 60], "ok": "Identifier"},
{"id": 19, "stage": "lexer", "source": [60, 61], "ok": "Operator"},
{"id": 20, "stage": "lexer", "source": [62, 65], "ok": "Primitive"},
{"id": 21, "stage": "lexer", "source": [66, 68], "ok": "Operator"},
{"id": 22, "stage": "lexer", "source": [69, 71], "ok": "Integer"},
{"id": 23, "stage": "lexer", "source": [71, 72], "ok": "Operator"},
{"id": 24, "stage": "lexer", "source": [77, 82], "ok": "Keyword"},
{"id": 25, "stage": "lexer", "source": [83, 84], "ok": "Operator"},
{"id": 26, "stage": "lexer", "source": [84, 85], "ok": "Identifier"},
{"id": 27, "stage": "lexer", "source": [86, 88], "ok": "Operator"},
{"id": 28, "stage": "lexer", "source": [89, 90], "ok": "Integer"},
{"id": 29, "stage": "lexer", "source": [91, 93], "ok": "Operator"},
{"id": 30, "stage": "lexer", "source": [94, 95], "ok": "Identifier"},
{"id": 31, "stage": "lexer", "source": [96, 98], "ok": "Operator"},
{"id": 32, "stage": "lexer", "source": [99, 101], "ok": "Integer"},
{"id": 33, "stage": "lexer", "source": [101, 102], "ok": "Operator"},
{"id": 34, "stage": "lexer", "source": [103, 104], "ok": "Operator"},
{"id": 35, "stage": "lexer", "source": [113, 120], "ok": "Keyword"},
{"id": 36, "stage": "lexer", "source": [120, 122], "ok": "Operator"},
{"id": 37, "stage": "lexer", "source": [122, 125], "ok": "Identifier"},
{"id": 38, "stage": "lexer", "source": [125, 127], "ok": "Operator"},
{"id": 39, "stage": "lexer", "source": [127, 129], "ok": "Identifier"},
{"id": 40, "stage": "lexer", "source": [129, 131], "ok": "Operator"},
{"id": 41, "stage": "lexer", "source": [131, 141], "ok": "Identifier"},
{"id": 42, "stage": "lexer", "source": [141, 142], "ok": "Operator"},
{"id": 43, "stage": "lexer", "source": [142, 143], "ok": "Identifier"},
{"id": 44, "stage": "lexer", "source": [143, 144], "ok": "Operator"},
{"id": 45, "stage": "lexer", "source": [144, 145], "ok": "Operator"},
{"id": 46, "stage": "lexer", "source": [154, 161], "ok": "Keyword"},
{"id": 47, "stage": "lexer", "source": [161, 163], "ok": "Operator"},
{"id": 48, "stage": "lexer", "source": [163, 166], "ok": "Identifier"},
{"id": 49, "stage": "lexer", "source": [166, 168], "ok": "Operator"},
{"id": 50, "stage": "lexer", "source": [168, 170], "ok": "Identifier"},
{"id": 51, "stage": "lexer", "source": [170, 172], "ok": "Operator"},
{"id": 52, "stage": "lexer", "source": [172, 182], "ok": "Identifier"},
{"id": 53, "stage": "lexer", "source": [182, 183], "ok": "Operator"},
{"id": 54, "stage": "lexer", "source": [183, 184], "ok": "Identifier"},
{"id": 55, "stage": "lexer", "source": [184, 185], "ok": "Operator"},
{"id": 56, "stage": "lexer", "source": [185, 186], "ok": "Operator"},
{"id": 57, "stage": "lexer", "source": [195, 198], "ok": "Keyword"},
{"id": 58, "stage": "lexer", "source": [199, 200], "ok": "Identifier"},
{"id": 59, "stage": "lexer", "source": [201, 203], "ok": "Operator"},
{"id": 60, "stage": "lexer", "source": [204, 205], "ok": "Identifier"},
{"id": 61, "stage": "lexer", "source": [206, 207], "ok": "Operator"},
{"id": 62, "stage": "lexer", "source": [208, 209], "ok": "Integer"},
{"id": 63, "stage": "lexer", "source": [209, 210], "ok": "Operator"},
{"id": 64, "stage": "lexer", "source": [219, 222], "ok": "Keyword"},
{"id": 65, "stage": "lexer", "source": [223, 224], "ok": "Identifier"},
{"id": 66, "stage": "lexer", "source": [225, 227], "ok": "Operator"},
{"id": 67, "stage": "lexer", "source": [228, 229], "ok": "Identifier"},
{"id": 68, "stage": "lexer", "source": [230, 231], "ok": "Operator"},
{"id": 69, "stage": "lexer", "source": [232, 233], "ok": "Integer"},
{"id": 70, "stage": "lexer", "source": [233, 234], "ok": "Operator"},
{"id": 71, "stage": "lexer", "source": [239, 240], "ok": "Operator"},
{"id": 72, "stage": "lexer", "source": [240, 241], "ok": "Operator"},
{"id": 73, "stage": "lexer", "source": [246, 252], "ok": "Keyword"},
{"id": 74, "stage": "lexer", "source": [253, 254], "ok": "Integer"},
{"id": 75, "stage": "lexer", "source": [254, 255], "ok": "Operator"},
{"id": 76, "stage": "lexer", "source": [256, 257], "ok": "Operator"},
{"id": 82, "parent_id": 80, "stage": "parser", "source": [16, 19], "ok": "Primitive Type"},
{"id": 80, "parent_id": 79, "stage": "parser", "source": [0, 19], "ok": "Routine Declaration"},
{"id": 86, "parent_id": 85, "stage": "parser", "source": [37, 40], "ok": "Primitive Type"},
{"id": 85, "parent_id": 84, "stage": "parser", "source": [34, 40], "ok": "Identifier Declaration"},
{"id": 97, "parent_id": 93, "stage": "parser", "source": [44, 45], "ok": "Number"},
{"id": 93, "parent_id": 92, "stage": "parser", "source": [44, 45]},
{"id": 92, "parent_id": 91, "stage": "parser", "source": [44, 45]},
{"id": 91, "parent_id": 90, "stage": "parser", "source": [44, 45]},
{"id": 90, "parent_id": 89, "stage": "parser", "source": [44, 45]},
{"id": 89, "parent_id": 88, "stage": "parser", "source": [44, 45]},
{"id": 88, "parent_id": 84, "stage": "parser", "source": [44, 45]},
{"id": 84, "parent_id": 83, "stage": "parser", "source": [26, 45], "ok": "Let Binding"},
{"id": 83, "parent_id": 79, "stage": "parser", "source": [26, 46], "ok": "Statement Bind"},
{"id": 101, "parent_id": 100, "stage": "parser", "source": [62, 65], "ok": "Primitive Type"},
{"id": 100, "parent_id": 99, "stage": "parser", "source": [59, 65], "ok": "Identifier Declaration"},
{"id": 112, "parent_id": 108, "stage": "parser", "source": [69, 71], "ok": "Number"},
{"id": 108, "parent_id": 107, "stage": "parser", "source": [69, 71]},
{"id": 107, "parent_id": 106, "stage": "parser", "source": [69, 71]},
{"id": 106, "parent_id": 105, "stage": "parser", "source": [69, 71]},
{"id": 105, "parent_id": 104, "stage": "parser", "source": [69, 71]},
{"id": 104, "parent_id": 103, "stage": "parser", "source": [69, 71]},
{"id": 103, "parent_id": 99, "stage": "parser", "source": [69, 71]},
{"id": 99, "parent_id": 98, "stage": "parser", "source": [51, 71], "ok": "Let Binding"},
{"id": 98, "parent_id": 79, "stage": "parser", "source": [51, 72], "ok": "Statement Bind"},
{"id": 131, "parent_id": 130, "stage": "parser", "source": [84, 85], "ok": "Path"},
{"id": 130, "parent_id": 128, "stage": "parser", "source": [84, 85], "ok": "Identifier"},
{"id": 128, "parent_id": 127, "stage": "parser", "source": [84, 85]},
{"id": 127, "parent_id": 126, "stage": "parser", "source": [84, 85]},
{"id": 126, "parent_id": 125, "stage": "parser", "source": [84, 85]},
{"id": 140, "parent_id": 136, "stage": "parser", "source": [89, 90], "ok": "Number"},
{"id": 136, "parent_id": 135, "stage": "parser", "source": [89, 90]},
{"id": 135, "parent_id": 134, "stage": "parser", "source": [89, 90]},
{"id": 134, "parent_id": 133, "stage": "parser", "source": [89, 90]},
{"id": 133, "parent_id": 125, "stage": "parser", "source": [89, 90]},
{"id": 125, "parent_id": 124, "stage": "parser", "source": [84, 90], "ok": "<="},
{"id": 148, "parent_id": 147, "stage": "parser", "source": [94, 95], "ok": "Path"},
{"id": 147, "parent_id": 145, "stage": "parser", "source": [94, 95], "ok": "Identifier"},
{"id": 145, "parent_id": 144, "stage": "parser", "source": [94, 95]},
{"id": 144, "parent_id": 143, "stage": "parser", "source": [94, 95]},
{"id": 143, "parent_id": 142, "stage": "parser", "source": [94, 95]},
{"id": 157, "parent_id": 153, "stage": "parser", "source": [99, 101], "ok": "Number"},
{"id": 153, "parent_id": 152, "stage": "parser", "source": [99, 101]},
{"id": 152, "parent_id": 151, "stage": "parser", "source": [99, 101]},
{"id": 151, "parent_id": 150, "stage": "parser", "source": [99, 101]},
{"id": 150, "parent_id": 142, "stage": "parser", "source": [99, 101]},
{"id": 142, "parent_id": 141, "stage": "parser", "source": [94, 101], "ok": ">="},
{"id": 141, "parent_id": 124, "stage": "parser", "source": [94, 101]},
{"id": 124, "parent_id": 123, "stage": "parser", "source": [84, 101], "ok": "&&"},
{"id": 123, "parent_id": 122, "stage": "parser", "source": [84, 101]},
{"id": 170, "parent_id": 169, "stage": "parser", "source": [113, 141], "ok": "Path"},
{"id": 179, "parent_id": 178, "stage": "parser", "source": [142, 143], "ok": "Path"},
{"id": 178, "parent_id": 176, "stage": "parser", "source": [142, 143], "ok": "Identifier"},
{"id": 176, "parent_id": 175, "stage": "parser", "source": [142, 143]},
{"id": 175, "parent_id": 174, "stage": "parser", "source": [142, 143]},
{"id": 174, "parent_id": 173, "stage": "parser", "source": [142, 143]},
{"id": 173, "parent_id": 172, "stage": "parser", "source": [142, 143]},
{"id": 172, "parent_id": 171, "stage": "parser", "source": [142, 143]},
{"id": 171, "parent_id": 169, "stage": "parser", "source": [142, 143]},
{"id": 169, "parent_id": 167, "stage": "parser", "source": [113, 144], "ok": "Routine Call"},
{"id": 167, "parent_id": 166, "stage": "parser", "source": [113, 144]},
{"id": 166, "parent_id": 165, "stage": "parser", "source": [113, 144]},
{"id": 165, "parent_id": 164, "stage": "parser", "source": [113, 144]},
{"id": 164, "parent_id": 163, "stage": "parser", "source": [113, 144]},
{"id": 163, "parent_id": 162, "stage": "parser", "source": [113, 144]},
{"id": 162, "parent_id": 159, "stage": "parser", "source": [113, 144]},
{"id": 159, "parent_id": 158, "stage": "parser", "source": [113, 145], "ok": "Statement Expression"},
{"id": 192, "parent_id": 191, "stage": "parser", "source": [154, 182], "ok": "Path"},
{"id": 201, "parent_id": 200, "stage": "parser", "source": [183, 184], "ok": "Path"},
{"id": 200, "parent_id": 198, "stage": "parser", "source": [183, 184], "ok": "Identifier"},
{"id": 198, "parent_id": 197, "stage": "parser", "source": [183, 184]},
{"id": 197, "parent_id": 196, "stage": "parser", "source": [183, 184]},
{"id": 196, "parent_id": 195, "stage": "parser", "source": [183, 184]},
{"id": 195, "parent_id": 194, "stage": "parser", "source": [183, 184]},
{"id": 194, "parent_id": 193, "stage": "parser", "source": [183, 184]},
{"id": 193, "parent_id": 191, "stage": "parser", "source": [183, 184]},
{"id": 191, "parent_id": 189, "stage": "parser", "source": [154, 185], "ok": "Routine Call"},
{"id": 189, "parent_id": 188, "stage": "parser", "source": [154, 185]},
{"id": 188, "parent_id": 187, "stage": "parser", "source": [154, 185]},
{"id": 187, "parent_id": 186, "stage": "parser", "source": [154, 185]},
{"id": 186, "parent_id": 185, "stage": "parser", "source": [154, 185]},
{"id": 185, "parent_id": 184, "stage": "parser", "source": [154, 185]},
{"id": 184, "parent_id": 181, "stage": "parser", "source": [154, 185]},
{"id": 181, "parent_id": 158, "stage": "parser", "source": [154, 186], "ok": "Statement Expression"},
{"id": 214, "parent_id": 213, "stage": "parser", "source": [204, 205], "ok": "Path"},
{"id": 213, "parent_id": 211, "stage": "parser", "source": [204, 205], "ok": "Identifier"},
{"id": 211, "parent_id": 210, "stage": "parser", "source": [204, 205]},
{"id": 210, "parent_id": 209, "stage": "parser", "source": [204, 205]},
{"id": 222, "parent_id": 218, "stage": "parser", "source": [208, 209], "ok": "Number"},
{"id": 218, "parent_id": 217, "stage": "parser", "source": [208, 209]},
{"id": 217, "parent_id": 216, "stage": "parser", "source": [208, 209]},
{"id": 216, "parent_id": 209, "stage": "parser", "source": [208, 209]},
{"id": 209, "parent_id": 208, "stage": "parser", "source": [204, 209], "ok": "+"},
{"id": 208, "parent_id": 207, "stage": "parser", "source": [204, 209]},
{"id": 207, "parent_id": 206, "stage": "parser", "source": [204, 209]},
{"id": 206, "parent_id": 205, "stage": "parser", "source": [204, 209]},
{"id": 205, "parent_id": 203, "stage": "parser", "source": [195, 209], "ok": "Mutate"},
{"id": 203, "parent_id": 158, "stage": "parser", "source": [195, 210], "ok": "Statement Mutate"},
{"id": 234, "parent_id": 233, "stage": "parser", "source": [228, 229], "ok": "Path"},
{"id": 233, "parent_id": 231, "stage": "parser", "source": [228, 229], "ok": "Identifier"},
{"id": 231, "parent_id": 230, "stage": "parser", "source": [228, 229]},
{"id": 230, "parent_id": 229, "stage": "parser", "source": [228, 229]},
{"id": 242, "parent_id": 238, "stage": "parser", "source": [232, 233], "ok": "Number"},
{"id": 238, "parent_id": 237, "stage": "parser", "source": [232, 233]},
{"id": 237, "parent_id": 236, "stage": "parser", "source": [232, 233]},
{"id": 236, "parent_id": 229, "stage": "parser", "source": [232, 233]},
{"id": 229, "parent_id": 228, "stage": "parser", "source": [228, 233], "ok": "-"},
{"id": 228, "parent_id": 227, "stage": "parser", "source": [228, 233]},
{"id": 227, "parent_id": 226, "stage": "parser", "source": [228, 233]},
{"id": 226, "parent_id": 225, "stage": "parser", "source": [228, 233]},
{"id": 225, "parent_id": 223, "stage": "parser", "source": [219, 233], "ok": "Mutate"},
{"id": 223, "parent_id": 158, "stage": "parser", "source": [219, 234], "ok": "Statement Mutate"},
{"id": 158, "parent_id": 122, "stage": "parser", "source": [103, 240], "ok": "Expression Block"},
{"id": 122, "parent_id": 121, "stage": "parser", "source": [77, 240], "ok": "While"},
{"id": 121, "parent_id": 120, "stage": "parser", "source": [77, 240]},
{"id": 120, "parent_id": 119, "stage": "parser", "source": [77, 240]},
{"id": 119, "parent_id": 118, "stage": "parser", "source": [77, 240]},
{"id": 118, "parent_id": 117, "stage": "parser", "source": [77, 240]},
{"id": 117, "parent_id": 116, "stage": "parser", "source": [77, 240]},
{"id": 116, "parent_id": 113, "stage": "parser", "source": [77, 240]},
{"id": 113, "parent_id": 79, "stage": "parser", "source": [77, 241], "ok": "Statement Expression"},
{"id": 295, "parent_id": 291, "stage": "parser", "source": [253, 254], "ok": "Number"},
{"id": 291, "parent_id": 290, "stage": "parser", "source": [253, 254]},
{"id": 290, "parent_id": 289, "stage": "parser", "source": [253, 254]},
{"id": 289, "parent_id": 288, "stage": "parser", "source": [253, 254]},
{"id": 288, "parent_id": 287, "stage": "parser", "source": [253, 254]},
{"id": 287, "parent_id": 286, "stage": "parser", "source": [253, 254]},
{"id": 286, "parent_id": 285, "stage": "parser", "source": [253, 254]},
{"id": 285, "parent_id": 79, "stage": "parser", "source": [246, 254], "ok": "Return"},
{"id": 79, "parent_id": 77, "stage": "parser", "source": [0, 257], "ok": "Funtion Definition"},
{"id": 77, "stage": "parser", "source": [0, 257], "ok": "File Module"},
{"id": 298, "stage": "canonize-item-path", "source": [0, 257], "ok": "$complex_cond::complex_cond"},
{"id": 299, "stage": "canonize-item-path", "source": [0, 257], "ok": "$complex_cond::complex_cond::complex_cond"},
{"id": 300, "stage": "canonize-item-path", "source": [0, 257], "ok": "$complex_cond::complex_cond::my_main"},
{"id": 301, "stage": "type-resolver", "source": [44, 45], "ok": "i64"},
{"id": 302, "stage": "type-resolver", "source": [26, 46], "ok": "i64"},
{"id": 303, "stage": "type-resolver", "source": [69, 71], "ok": "i64"},
{"id": 304, "stage": "type-resolver", "source": [51, 72], "ok": "i64"},
{"id": 305, "stage": "type-resolver", "source": [84, 85], "ok": "i64", "ref": [26, 46]},
{"id": 306, "stage": "type-resolver", "source": [89, 90], "ok": "i64"},
{"id": 307, "stage": "type-resolver", "source": [84, 90], "ok": "bool"},
{"id": 308, "stage": "type-resolver", "source": [94, 95], "ok": "i64", "ref": [51, 72]},
{"id": 309, "stage": "type-resolver", "source": [99, 101], "ok": "i64"},
{"id": 310, "stage": "type-resolver", "source": [94, 101], "ok": "bool"},
{"id": 311, "stage": "type-resolver", "source": [84, 101], "ok": "bool"},
{"id": 312, "stage": "type-resolver", "source": [142, 143], "ok": "i64", "ref": [26, 46]},
{"id": 313, "stage": "type-resolver", "source": [113, 145], "ok": "Unit"},
{"id": 314, "stage": "type-resolver", "source": [183, 184], "ok": "i64", "ref": [51, 72]},
{"id": 315, "stage": "type-resolver", "source": [154, 186], "ok": "Unit"},
{"id": 316, "stage": "type-resolver", "source": [204, 205], "ok": "i64", "ref": [26, 46]},
{"id": 317, "stage": "type-resolver", "source": [208, 209], "ok": "i64"},
{"id": 318, "stage": "type-resolver", "source": [204, 209], "ok": "i64"},
{"id": 319, "stage": "type-resolver", "source": [195, 210], "ok": "i64"},
{"id": 320, "stage": "type-resolver", "source": [228, 229], "ok": "i64", "ref": [51, 72]},
{"id": 321, "stage": "type-resolver", "source": [232, 233], "ok": "i64"},
{"id": 322, "stage": "type-resolver", "source": [228, 233], "ok": "i64"},
{"id": 323, "stage": "type-resolver", "source": [219, 234], "ok": "i64"},
{"id": 324, "stage": "type-resolver", "source": [103, 240], "ok": "Unit"},
{"id": 325, "stage": "type-resolver", "source": [77, 241], "ok": "Unit"},
{"id": 326, "stage": "type-resolver", "source": [253, 254], "ok": "i64"},
{"id": 327, "stage": "type-resolver", "source": [246, 254], "ok": "i64"},
{"id": 328, "stage": "type-resolver", "source": [0, 257], "ok": "i64"},
{"id": 333, "parent_id": 332, "stage": "llvm", "source": [44, 45], "ok": "i64 0"},
{"id": 332, "parent_id": 331, "stage": "llvm", "source": [26, 46], "ok": "  %x = alloca i64, align 8"},
{"id": 331, "parent_id": 330, "stage": "llvm", "source": [26, 46], "ok": "  store i64 0, i64* %x, align 4"},
{"id": 330, "parent_id": 329, "stage": "llvm", "source": [26, 46], "ok": "  %x = alloca i64, align 8"},
{"id": 337, "parent_id": 336, "stage": "llvm", "source": [69, 71], "ok": "i64 15"},
{"id": 336, "parent_id": 335, "stage": "llvm", "source": [51, 72], "ok": "  %y = alloca i64, align 8"},
{"id": 335, "parent_id": 334, "stage": "llvm", "source": [51, 72], "ok": "  store i64 15, i64* %y, align 4"},
{"id": 334, "parent_id": 329, "stage": "llvm", "source": [51, 72], "ok": "  %y = alloca i64, align 8"},
{"id": 338, "parent_id": 329, "stage": "llvm", "source": [77, 241], "ok": "  br label %while_cond"},
{"id": 342, "parent_id": 341, "stage": "llvm", "source": [84, 85], "ok": "  %x1 = load i64, i64* %x, align 4"},
{"id": 343, "parent_id": 341, "stage": "llvm", "source": [89, 90], "ok": "i64 5"},
{"id": 341, "parent_id": 340, "stage": "llvm", "source": [84, 90], "ok": "  %0 = icmp sle i64 %x1, 5"},
{"id": 345, "parent_id": 344, "stage": "llvm", "source": [94, 95], "ok": "  %y2 = load i64, i64* %y, align 4"},
{"id": 346, "parent_id": 344, "stage": "llvm", "source": [99, 101], "ok": "i64 11"},
{"id": 344, "parent_id": 340, "stage": "llvm", "source": [94, 101], "ok": "  %1 = icmp sge i64 %y2, 11"},
{"id": 340, "parent_id": 339, "stage": "llvm", "source": [84, 101], "ok": "  %2 = and i1 %0, %1"},
{"id": 339, "parent_id": 329, "stage": "llvm", "source": [77, 241], "ok": "  br i1 %2, label %while_body, label %while_end"},
{"id": 349, "parent_id": 348, "stage": "llvm", "source": [142, 143], "ok": "  %x3 = load i64, i64* %x, align 4"},
{"id": 348, "parent_id": 347, "stage": "llvm", "source": [113, 145], "ok": "  call void @std_io_writei64ln(i64 %x3)"},
{"id": 351, "parent_id": 350, "stage": "llvm", "source": [183, 184], "ok": "  %y4 = load i64, i64* %y, align 4"},
{"id": 350, "parent_id": 347, "stage": "llvm", "source": [154, 186], "ok": "  call void @std_io_writei64ln(i64 %y4)"},
{"id": 354, "parent_id": 353, "stage": "llvm", "source": [204, 205], "ok": "  %x5 = load i64, i64* %x, align 4"},
{"id": 355, "parent_id": 353, "stage": "llvm", "source": [208, 209], "ok": "i64 1"},
{"id": 353, "parent_id": 352, "stage": "llvm", "source": [204, 209], "ok": "  %3 = add i64 %x5, 1"},
{"id": 352, "parent_id": 347, "stage": "llvm", "source": [195, 210], "ok": "  store i64 %3, i64* %x, align 4"},
{"id": 358, "parent_id": 357, "stage": "llvm", "source": [228, 229], "ok": "  %y6 = load i64, i64* %y, align 4"},
{"id": 359, "parent_id": 357, "stage": "llvm", "source": [232, 233], "ok": "i64 1"},
{"id": 357, "parent_id": 356, "stage": "llvm", "source": [228, 233], "ok": "  %4 = sub i64 %y6, 1"},
{"id": 356, "parent_id": 347, "stage": "llvm", "source": [219, 234], "ok": "  store i64 %4, i64* %y, align 4"},
{"id": 347, "parent_id": 329, "stage": "llvm", "source": [77, 241], "ok": "  br label %while_cond"},
{"id": 361, "parent_id": 360, "stage": "llvm", "source": [253, 254], "ok": "i64 0"},
{"id": 360, "parent_id": 329, "stage": "llvm", "source": [246, 254], "ok": "  ret i64 0"},
{"id": 329, "stage": "llvm", "source": [0, 257], "ok": "define i64 @complex_cond_complex_cond_my_main() {\nentry:\n  %x = alloca i64, align 8\n  store i64 0, i64* %x, align 4\n  %y = alloca i64, align 8\n  store i64 15, i64* %y, align 4\n  br label %while_cond\n\nwhile_cond:                                       ; preds = %while_body, %entry\n  %x1 = load i64, i64* %x, align 4\n  %0 = icmp sle i64 %x1, 5\n  %y2 = load i64, i64* %y, align 4\n  %1 = icmp sge i64 %y2, 11\n  %2 = and i1 %0, %1\n  br i1 %2, label %while_body, label %while_end\n\nwhile_body:                                       ; preds = %while_cond\n  %x3 = load i64, i64* %x, align 4\n  call void @std_io_writei64ln(i64 %x3)\n  %y4 = load i64, i64* %y, align 4\n  call void @std_io_writei64ln(i64 %y4)\n  %x5 = load i64, i64* %x, align 4\n  %3 = add i64 %x5, 1\n  store i64 %3, i64* %x, align 4\n  %y6 = load i64, i64* %y, align 4\n  %4 = sub i64 %y6, 1\n  store i64 %4, i64* %y, align 4\n  br label %while_cond\n\nwhile_end:                                        ; preds = %while_cond\n  ret i64 0\n}\n"}]