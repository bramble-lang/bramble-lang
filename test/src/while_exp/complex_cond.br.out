0
15
1
14
2
13
3
12
4
11
